adapters/cpmstar/cpmstar.go

```diff
@@ -77,14 +96,48 @@ func preprocess(request *openrtb2.BidRequest) error {
 			return err
 		}
 
+		// Validate the bidder-specific configuration
 		var extImp openrtb_ext.ExtImpCpmstar
 		if err := jsonutil.Unmarshal(bidderExt.Bidder, &extImp); err != nil {
 			return &errortypes.BadInput{
 				Message: err.Error(),
 			}
 		}
 
-		imp.Ext = bidderExt.Bidder
+		// Create new extension object that preserves all original fields except 'bidder'
+		newExt := make(map[string]json.RawMessage)
+		for key, value := range originalExt {
+			if key != "bidder" {
+				newExt[key] = value
+			}
+		}
+
+		// Add bidder configuration fields directly to the root level
+		var bidderConfig map[string]interface{}
+		if err := jsonutil.Unmarshal(bidderExt.Bidder, &bidderConfig); err != nil {
+			return &errortypes.BadInput{
+				Message: err.Error(),
+			}
+		}
+
+		for key, value := range bidderConfig {
+			valueBytes, err := json.Marshal(value)
+			if err != nil {
+				return &errortypes.BadInput{
+					Message: err.Error(),
+				}
+			}
+			newExt[key] = valueBytes
```

Comment:
Can we avoid double unmarshal of the same `bidderExt.Bidder` content (once as `map`, once as `struct`) by converting to map of json.RawMessage for direct re-attachment.

```
var bidderConfig map[string]json.RawMessage
if err := jsonutil.Unmarshal(bidderExt.Bidder, &bidderConfig); err != nil {
      return &errortypes.BadInput{Message: "Invalid bidder nested structure: " + err.Error()}
}

for k, v := range bidderConfig {
      newExt[k] = v
}
```